sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -262,13 +344,14 @@ public virtual async Task StopAsync()
                 }
                 finally
                 {
+                    RunningTask = null;
```

Comment:
Just moved it here to make sure we make it `null` even if something unexpected fails when stopping. We need to do that if we want to make sure we are able to start again. Not sure if it's better to make it `null` only when we ensure we could stop properly.

Reply:
To stop, we should await the task and when that is complete, set it to null.   (success or failure)  no?

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/769b2a8cd8c38b82933b2286f56dc3c4e79bc4c4/sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs#L328-L350

That's kind of what we're doing right now, but I'm making sure we set it `null` only **after** closing all pumps. Otherwise we would have a race condition where the user could update one of handlers while some pumps are still running (because `RunningTask = null`). We could wrap the `StopAsync` method in a lock as well, as an alternative, but I think the way it is it's cleaner.